browser/brave_wallet/keyring_service_unittest.cc

```diff
@@ -1967,7 +1977,8 @@ TEST_F(KeyringServiceUnitTest, RestoreLegacyBraveWallet) {
   KeyringService service(json_rpc_service(), GetPrefs(), GetLocalState());
   BraveWalletService brave_wallet_service(
       shared_url_loader_factory(), nullptr, &service, json_rpc_service(),
-      nullptr, nullptr, nullptr, GetPrefs(), GetLocalState(), false);
+      tx_service(&service).get(), nullptr, nullptr, GetPrefs(), GetLocalState(),
```

Comment:
This creates `unique_ptr<TxService>` and immediately destroys it leaving danglig pointer in `brave_wallet_service.tx_service_` -- most likely will crash in ASAN builds. It should be either TxService living on stack like `KeyringService service` above or allowing tx_service to be null for tests(see `CHECK_IS_TEST()`)

Reply:
TIL about `CHECK_IS_TEST` 👍 . I went with `MakeTxService` route given [check_is_test.h documentation](https://source.chromium.org/chromium/chromium/src/+/main:base/check_is_test.h;l=14-15?q=CHECK_IS_TEST).